api/ombpdf/footnotes.py

```diff
@@ -116,19 +118,14 @@ def finish_footnote():
                     if found_horiz_line:
                         # Here we pull out the previous footnote and continue
                         # adding to it.
-                        curr_ft = footnotes.pop()
-                        curr_obj = curr_ft[0]
-                        curr_num = curr_obj[0]
-                        curr_txt = curr_obj[1]
-                        curr_lines = curr_ft[1]
+                        (curr_num,), curr_txt, curr_lines = footnotes.pop()
                         curr_footnote = [curr_num, curr_txt + chars, curr_lines
                                          + [line]]
```

Comment:
Only tangential to this PR, but should we use the same data type (now a triple, see `footnote_plus`) here rather than making a new type here (a three-element list)? Looks like it's mutated ~115, but that could be an namedtuple `._replace()`

Reply:
Yeah, this code is not very nice, blurg.  I would like to use a named tuple here but I feel weird about making named tuples for intermediate data structures that are used internally by functions... Then again, this whole function is a bit over-long and could be broken up into multiple ones, and improving readability is always a good thing.

I did not know that `_replace()` existed, thanks for suggesting it!